.github/workflows/build.yml

```diff
@@ -95,6 +95,46 @@ jobs:
         shell: bash
         run: |
           pytest -sv tests/unit
+      - name: Check for Secret availability
+        id: secret-check
+        if: ${{ contains(fromJSON('["3.9"]'), matrix.python) }}
+        # perform secret check & put boolean result as an output
+        shell: bash
+        run: |
+          if [ -z "${{ secrets.encrypted_d17283647768_key }}" ]  || [ -z "${{ secrets.encrypted_d17283647768_iv }}" ]; then
+            echo "secrets_available=false" >> $GITHUB_OUTPUT;
+          else
+            echo "secrets_available=true" >> $GITHUB_OUTPUT;
+          fi
+      - name: Install AWS Tools
+        id: aws-check
+        if: ${{ steps.secret-check.outputs.secrets_available == 'true' && !(startsWith(matrix.os, 'windows')) }}
+        shell: bash
+        run: |
+          echo "run_opentelemetry=true" >> $GITHUB_OUTPUT;
+
+          # AWS CLI is pre-installed on github hosted runners
+          # curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          # unzip awscliv2.zip
+          # sudo ./aws/install
+          # curl "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/ubuntu_64bit/session-manager-plugin.deb" -o "session-manager-plugin.deb"
+          # sudo dpkg -i session-manager-plugin.deb
```

Comment:
I needed it for running the pipeline on my local machine because it didn't come packages with the CLI tools already installed. But GH runners already have AWS CLI tools installed